probing/gap_preprocess.py

```diff
@@ -0,0 +1,137 @@
+import retokenize_edge_data as retokenize
+import sys
+import argparse
+import pandas as pd
+import pickle
+from enum import Enum
+
+
+UNKNOWN = "Unknown"
+MASCULINE = "Male"
+FEMININE = "Female"
+
+
+# Mapping of (lowercased) pronoun form to gender value. Note that reflexives
+# are not included in GAP, so do not appear here.
+PRONOUNS = {
+    'she': FEMININE,
+    'her': FEMININE,
+    'hers': FEMININE,
+    'he': MASCULINE,
+    'his': MASCULINE,
+    'him': MASCULINE,
+}
+
+def getEnd(index, word):
+    return index  + len(word)
+
+def build_spans_aligned_with_tokenization(text, tokenizer_name, orig_span1_start, orig_span1_end, orig_span2_start, orig_span2_end):
+    """
+    Builds the alignment while also tokenizing the input piece by piece. 
+    """
+    if orig_span1_end > orig_span2_start:
+        # switch them since the pronoun comes after 
+        span2_start = orig_span1_start
+        span2_end = orig_span1_end
+        span1_start = orig_span2_start
+        span1_end = orig_span2_end
+    else:
+        span1_start = orig_span1_start
+        span1_end = orig_span1_end
+        span2_start = orig_span2_start
+        span2_end = orig_span2_end
+
+    current_tokenization = []
+    aligner_fn = retokenize.get_aligner_fn(tokenizer_name)
+    text1 = text[:span1_start]
+    ta, new_tokens = aligner_fn(text1)
+    current_tokenization.extend(new_tokens)
+    new_span1start = len(current_tokenization)
+
+    span1 = text[span1_start:span1_end] 
+    ta, span_tokens = aligner_fn(span1)
+    current_tokenization.extend(span_tokens)
+    new_span1end = len(current_tokenization) 
+    text2 = text[span1_end:span2_start]
+    ta, new_tokens = aligner_fn(text2)
+    current_tokenization.extend(new_tokens)
+    new_span2start = len(current_tokenization)
+
+    span2 = text[span2_start:span2_end]
+    ta, span_tokens = aligner_fn(span2)
+    current_tokenization.extend(span_tokens)
+    new_span2end = len(current_tokenization)
+
+    text3 = text[span2_end:]
+    ta, span_tokens = aligner_fn(text3)
+    current_tokenization.extend(span_tokens)
+
+    text = " ".join(current_tokenization)
+    if orig_span1_end > orig_span2_start:
+        return new_span2start, new_span2end, new_span1start, new_span1end,text
+    return  new_span1start, new_span1end, new_span2start, new_span2end, text
+
+def align_spans(split, tokenizer_name, data_dir):
+    """
+        This processes the dataset into the form that edge probing can read in
```

Comment:
This isn't a probing task. If we reuse code that's currently framed as being specific to edge probing, we should move or rename that existing code to make it clear that it's just about _edges_, not _probing_.

Reply:
Again this will be in a separate PR coming up in the next few days - I just wanted to get GAP benchlines started asap with this PR